src/magic_folder/test/cli/test_magic_folder.py

```diff
@@ -70,98 +80,111 @@ def setUp(self):
+
+        class GlobalService(object):
+            def get_folder_service(s, name):
+                return self.magic_folders[name]
+
+        self.config = create_testing_configuration(
+            FilePath(self.mktemp()),
+            self.node_directory,
+            GlobalService(),
+        )
 
     @defer.inlineCallbacks
     def test_list_none(self):
         """
         When there are no Magic Folders at all, the output of the list command
         reports this.
         """
-        outcome = yield cli(
-            self.config_dir,
-            [b"list"],
+        output = StringIO()
+        yield magic_folder_list(self.config, output)
+        self.assertThat(
+            output.getvalue(),
+            Contains(u"No magic-folders")
         )
-        self.assertThat(outcome.stdout, Contains(u"No magic-folders"))
 
     @defer.inlineCallbacks
     def test_list_none_json(self):
         """
         When there are no Magic Folders at all, the output of the list command
         reports this in JSON format if given ``--json``.
         """
-        outcome = yield cli(
-            self.config_dir,
-            [b"list", b"--json"],
+        output = StringIO()
+        yield magic_folder_list(self.config, output, as_json=True)
+        self.assertThat(
+            output.getvalue(),
+            AfterPreprocessing(json.loads, Equals({}))
         )
-        self.assertThat(outcome.stdout, AfterPreprocessing(json.loads, Equals({})))
 
     @defer.inlineCallbacks
     def test_list_some(self):
         """
         When there are Magic Folders, the output of the list command describes
         them.
         """
-        # Get a magic folder.
-        folder_path = self.tempdir.child(u"magic-folder")
+        folder_path = FilePath(self.mktemp())
         folder_path.makedirs()
 
-        outcome = yield cli(
-            self.config_dir, [
-                b"add",
-                b"--name", b"list-some-folder",
-                b"--author", b"alice",
-                folder_path.asBytesMode().path,
-            ],
-        )
-        self.assertThat(
-            outcome.succeeded(),
-            Equals(True),
+        self.config.create_magic_folder(
+            u"list-some-folder",
+            folder_path,
+            folder_path.child(u".state"),
+            create_local_author(u"alice"),
+            u"URI:DIR2-RO:ou5wvazwlyzmqw7yof5ifmgmau:xqzt6uoulu4f3m627jtadpofnizjt3yoewzeitx47vw6memofeiq",
+            u"URI:DIR2:bgksdpr3lr2gvlvhydxjo2izea:dfdkjc44gg23n3fxcxd6ywsqvuuqzo4nrtqncrjzqmh4pamag2ia",
+            1,
         )
 
-        outcome = yield cli(
-            self.config_dir,
-            [b"list"],
-        )
-        self.expectThat(outcome.stdout, Contains(b"list-some-folder"))
-        self.expectThat(outcome.stdout, Contains(folder_path.path))
+        output = StringIO()
+        yield magic_folder_list(self.config, output)
+        self.expectThat(output.getvalue(), Contains(b"list-some-folder"))
```

Comment:
I think looking for `b"..."` in a `StringIO` only works until there's any non-ASCII anywhere.  I'm inclined to suggest `BytesIO` here but same note as above about not unnecessarily expanding this PR's scope...